app/models/manageiq/providers/ibm_terraform/inventory/parser/configuration_manager.rb

```diff
@@ -1,18 +1,20 @@
 class ManageIQ::Providers::IbmTerraform::Inventory::Parser::ConfigurationManager < ManageIQ::Providers::IbmTerraform::Inventory::Parser
   def parse
-    configuration_profiles
-    configured_systems
+    persister_profiles = configuration_profiles
+    configured_systems(persister_profiles)
   end
 
   def configuration_profiles
-    collector.templates.each do |template|
+    configuration_profiles = collector.templates.map do |template|
       persister.configuration_profiles.build(
         :manager_ref     => template["id"].to_s,
         :name            => template["name"],
         :description     => template["description"],
         :target_platform => template.dig("manifest", "template_provider")
       )
     end
+
+    configuration_profiles
```

Comment:
`collector.templates.map` will automatically return the array so you can skip the local variable plus the return and just do

```ruby
def configuration_profiles
  configuration_profiles = collector.templates.map do |template|
    ...
  end
end
```